server/athenian/api/internal/features/entries.py

```diff
@@ -130,6 +130,38 @@ def _postprocess_cached_facts(
     return result
 
 
+class _PRMetricsLineGHCache:
+    @staticmethod
+    def postprocess(result: np.ndarray, *, metrics: Sequence[str], **kwargs: Any) -> np.ndarray:
+        # cache value has been stored with sorted(metrics), adapt before returning to caller
+        if (sorted_metrics := sorted(metrics)) != metrics:
+            sort_indexes = [metrics.index(m) for m in sorted_metrics]
+            return _sorted_result_metrics(result, sort_indexes)
+
+        return result
+
+    @staticmethod
+    def preprocess(result: np.ndarray, *, metrics: Sequence[str], **kwargs: Any) -> np.ndarray:
+        # cache value must be stored with sorted(metrics), adapt if needed
+        if (sorted_metrics := sorted(metrics)) != metrics:
+            sort_indexes = [sorted_metrics.index(m) for m in metrics]
+            return _sorted_result_metrics(result, sort_indexes)
+        return result
+
+
+def _sorted_result_metrics(result: np.ndarray, metrics_sorted_indexes: list[int]) -> np.ndarray:
+    def _sort_level(lst, depth: int, sort_indexes: list):
+        if depth <= 0:
+            elements = [lst[idx] for idx in sort_indexes]
+            if isinstance(lst, np.ndarray):  # keep np array as np array
+                return np.array(elements)
+            return elements
+        return [_sort_level(sublst, depth - 1, sort_indexes) for sublst in lst]
+
+    # list of metrics to sort is the sixth dimension
+    return _sort_level(result, 5, metrics_sorted_indexes)
```

Comment:
Won't this work?
```suggestion
    return result[: ,:, :, :, :, metrics_sorted_indexes]
```

Reply:
I tried to do everything with numpy but problem is that ndarray has 4 axis, the last two dimensions are python lists. This is because "time intervals" for each granularity is variable.

Reply:
I see. Then this code:
```python
for obj in result.ravel():
    for m in obj:
        m[:] = (m[idx] for idx in sort_indexes)
```

Reply:
tried a variant since I need to preserve result